pkg/labels/consul_aggregator.go

```diff
@@ -111,6 +151,11 @@ func (c *consulAggregator) removeWatch(watch *selectorWatch) {
 	} else {
 		c.watchers.delete(watch)
 	}
+	if c.watchers != nil {
+		c.metWatchCount.Update(int64(c.watchers.len()))
+	} else {
+		c.metWatchCount.Update(0)
```

Comment:
Because you asked me about it - yes it would have been safe http://play.golang.org/p/6uR_1sgNRT

My understanding of the situation is even though it is nil, the nil is a typed nil, therefore you can call methods on it, so long as the methods do not attempt to deref the nil (which `len` does not).

The case I usually use an example for this is https://github.com/golang/crypto/blob/master/openpgp/packet/config.go where the config checks if it's nil so that calling these methods on a nil config gives you the defaults.

Interesting but I'm not a strong advocate for changing the code since as written it's more obvious.